File: libs/baselib/src/main/java/com/base/framwork/fragment/BaseFragment.java
package com.base.framwork.fragment;

import android.content.Context;
import android.content.Intent;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.CallSuper;
import androidx.annotation.IdRes;
import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;

import com.base.framwork.view.ILifeProcessor;

import org.jetbrains.annotations.NotNull;

/**
 * @date 2019-12-09
 * @Author luffy
 * @description
 */
public abstract class BaseFragment extends Fragment implements ILifeProcessor {

    private View rootView;

    @Override
    public void onAttach(@NonNull Context context) {
        super.onAttach(context);
    }

    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        initParams(null);
    }

    /**
     * @param intent 在fragment 中总是null
     */
    public void initParams(@NotNull Intent intent) {

    }

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        super.onCreateView(inflater,container,savedInstanceState);
        if (generateIdLayout() > 0) {
            rootView = inflater.inflate(generateIdLayout(),container,false);
        } else if (generateViewLayout() != null) {
            rootView = generateViewLayout();
        }
        setStatusBar();
        initView();
        initEvent();
        initData();
        return rootView;
    }

    @Override
    public void onResume() {
        super.onResume();
    }

    @Override
    public void onPause() {
        super.onPause();
    }

    @NotNull
    @Override
    public View generateViewLayout() {
        return null;
    }

    @Override
    @CallSuper
    public void onDestroyView() {
        super.onDestroyView();
        rootView = null;
    }

    @Override
    public void setStatusBar() {
        // empty
        // 一般activity 已经有实现了，如果单独fragment需要，重写
    }

//    /**
//     * 用新的Androidx新的懒加载方案，旧方案用{@link AbstractLazyLoadFragment}
//     * 不过旧方案都过时了，google不建议使用
//     * 是否懒加载，不需要重写
//     * @return 默认为懒加载
//     */
//    public boolean isLazyLoad() {
//        return true;
//    }

    /**
     * 提供给java 用，尽早用kotlin吧
     * @param id
     * @param <T>
     * @return
     */
    public <T extends View> T findViewById(@IdRes int id) {
        return rootView.findViewById(id);
    }

}

File: libs/baselib/src/main/java/com/base/framwork/fragment/BaseFragment.kt
package com.base.framwork.fragment

import android.os.Bundle
import android.view.View
import androidx.annotation.CallSuper
import androidx.annotation.IdRes
import androidx.fragment.app.Fragment
import androidx.lifecycle.LifecycleObserver
import androidx.lifecycle.ViewModel
import com.base.framwork.activity.BaseActivity

/**
 * @date 2019-12-08
 * @Author luffy
 * @description 同[BaseActivity]
 */
open class BaseFragment : Fragment(){

    var isLoaded = false
    var rootView: View? = null

    @CallSuper
    override fun onResume() {
        super.onResume()
        if (!isLoaded) {
            lazyLoad()
            isLoaded = true
        }
        // 有的需要后续可见的时候
        if (isLoaded) {
            afterLazyLoad()
        }
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        rootView = view
        // 防止事件穿透
        view.isClickable = true
        super.onViewCreated(view, savedInstanceState)
    }


    /**
     * 显示dialog
     */
//    open fun showLoadingDialog(): LoadingDialog? {
//        return LoadingDialogHelper.instance.showLoading(activity, getLoadingAnimateSrc())
//    }

    /**
     * 隐藏dialog
     *
     */
    open fun hideLoadingDialog() {
//        LoadingDialogHelper.instance.hideLoading(activity)
    }

    /***
     * 定制当前Activity loading动画资源
     *
     * @return lottie json资源，assets中的文件资源全名：如**yello.json** <br></br>
     * 如果返回空则使用默认数据,见[LoadingDialog]
     */
    open fun getLoadingAnimateSrc(): String? {
        return null
    }


    @CallSuper
    override fun onDestroyView() {
        super.onDestroyView()
        isLoaded = false
        rootView = null

    }

    /**
     * 第一次可见，调用，一般用于懒加载
     */
    open fun lazyLoad() {

    }

    /**
     * 在第一次懒加载之后，如果切换resumue 可见需要刷新数据的情况
     */
    open fun afterLazyLoad() {

    }

    /**
     * 有轮询任务的时候复写
     */
    open fun refreshByPoll() {

    }
}